### DISCUSSION TITLE
Property '$queryRaw' does not exist on type 'PrismaClient<PrismaClientOptions, never, DefaultArgs>'.t

### DISCUSSION THREAD

**Discussion by MarvinTchio** [prisma/prisma#23910] [Q&A]
### Question

I am following CloudBeds MyAllocator Api documentations and given the `healthCheck.ts` route below, I am not able to set up the result because `queryRaw` doesn't exist but I would expect it too  given I have prisma v5.12;

```ts
import { NextApiRequest, NextApiResponse } from 'next';
import prisma from "@/app/libs/prismadb";  

export default async function handleHealthCheck(req: NextApiRequest, res: NextApiResponse) {
    try {
        // Optionally check database connectivity
        const result = await prisma.$queryRaw`SELECT 1`;  // Simple query to check database responsiveness
        if (result) {
            // If the query succeeds, return a success response
            return res.status(200).json({ success: true, message: 'API and database are operational.' });
        }
    } catch (error) {
        // Log the error and return a failure response if the database query fails
        console.error("Health Check Failed: ", error);
        return res.status(500).json({ success: false, message: "API or database is not operational." });
    }
}
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model OtaProperty {
  id                  String        @id @default(auto()) @map("_id") @db.ObjectId
  otaPropertyId       String        @unique
  otaPropertyPassword String
  sharedSecret        String
  roomTypes           RoomType[]
  ratePlans           RatePlan[]
  rooms               Room[]
  bookings            Booking[]
  listingId           String?       @unique
  listing             Listing?      @relation(fields: [listingId], references: [id])

  // Hierarchical relationships
  parentId            String?       @db.ObjectId
  subProperties       OtaProperty[] @relation("PropertyHierarchy")
  parentProperty      OtaProperty?  @relation("PropertyHierarchy", fields: [parentId], references: [id], onDelete: NoAction, onUpdate: NoAction)
}

model RoomType {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  name      String
  capacity  Int
  property  OtaProperty @relation(fields: [propertyId], references: [id])
  propertyId String
}

model RatePlan {
  id        String     @id @default(auto()) @map("_id") @db.ObjectId
  name      String
  price     Float
  currency  String
  property  OtaProperty @relation(fields: [propertyId], references: [id])
  propertyId String
}

model Room {
  id          String      @id @default(auto()) @map("_id") @db.ObjectId
  property    OtaProperty @relation(fields: [propertyId], references: [id])
  propertyId  String
  rate        Float       // Assuming rate is a float value
  available   Boolean     // Room availability status
}


model Booking {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  propertyId  String
  customerDetails String
  updatedAt   DateTime
  status             String? 
  cancellationReason String?
  property    OtaProperty @relation(fields: [propertyId], references: [id])
}
```

```ts
// Add any relevant Prisma Client queries here
```
`prisma.$queryRaw`SELECT 1``

- OS: .     `macOS`
- Database: . `MongoDB `
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it
`Environment variables loaded from .env
prisma                  : 5.12.1
@prisma/client          : 5.12.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.3.0
Query Engine (Node-API) : libquery-engine 473ed3124229e22d881cb7addf559799debae1ab (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 473ed3124229e22d881cb7addf559799debae1ab (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.12.0-21.473ed3124229e22d881cb7addf559799debae1ab
Default Engines Hash    : 473ed3124229e22d881cb7addf559799debae1ab
Studio                  : 0.499.0`

**Comment by ludralph** [comment]
Hi @MarvinTchio 👋 

Thank you for raising this question.

Can you ensure that the `prisma` import on this line `import prisma from "@/app/libs/prismadb"; ` is an instance of Prisma Client if it isn't?

 If it is, then can you try restarting your Typescript server or your IDE. You may also want to share the contents of the `@/app/libs/prismadb` file as well so we can take a look.

**Reply by MarvinTchio** [reply]
This route works as expected : 
```
import { NextApiRequest, NextApiResponse } from 'next';
import prisma from "@/app/libs/prismadb";

export default async function handleGetSubProperties(req: NextApiRequest, res: NextApiResponse) {
    const { ota_property_id } = req.body;

    if (!ota_property_id) {
        return res.status(400).json({ success: false, message: "Missing OTA property ID" });
    }

    try {
        const mainProperty = await prisma.otaProperty.findUnique({
            where: { otaPropertyId: ota_property_id },
            include: { subProperties: true }  
        });

        if (!mainProperty) {
            return res.status(404).json({ success: false, message: "Main property not found" });
        }

        // Return the list of sub-properties
        return res.status(200).json({ success: true, subProperties: mainProperty.subProperties });
    } catch (error) {
        console.error("Error fetching sub-properties:", error);
        return res.status(500).json({ success: false, message: "Internal server error" });
    }
}
```

where I also call:

```

 generator client {
  provider = "prisma-client-js"
}

```

from prismadb.
I'm not sure why this works and the health check route doesn't.

**Reply by ludralph** [reply]
Hi @MarvinTchio 

To better understand and resolve the issue with your health check route, it would be helpful to see its implementation.  If you're still facing issues, you might want to consider implementing a simple database query in your health check route.
```
await prisma.$queryRaw`SELECT 1`;
```
This simple query can help verify if the basic database connection is working in your health check route.

**Reply by viktordanko** [reply]
The **mongodb** provider does not support `$queryRaw`.
Try to use `$runCommandRaw` instead https://www.prisma.io/docs/orm/prisma-client/using-raw-sql/raw-queries#raw-queries-with-mongodb

**Reply by ludralph** [reply]
You're correct about the use of `runCommandRaw` instead of `queryRaw`. Thanks for pointing that out.

**Reply by MarvinTchio** [reply]
This issue can be closed. Fixed it a long time ago and forgot how by now, really appreciate this community for helping as best it could!

**Reply by Nditah** [reply]
So resolve int issue `Property '$queryRaw' does not exist on type 'PrismaService'.` in Nestjs 

```ts
await this.prisma.$runCommandRaw({ ping: 1 }); // for mongodb
await this.prisma.$queryRaw`SELECT 1`; // for RDBMs
```
